### PR TITLE
Feature/np 2470 update user info

### PR DESCRIPTION
**Pull request by gaizkadc**
#### What does this PR do?
UpdateUser updated.
#### What are the relevant tickets?

- [NP-2470](https://nalej.atlassian.net/browse/NP-2470)

### CODE REVIEW

File: internal/app/cli/users.go [hunk 1]
```diff
@@ -173,6 +173,14 @@ func (u *Users) Update(organizationID string, email string, newName string) {
 	updateRequest := &grpc_user_go.UpdateUserRequest{
 		OrganizationId: organizationID,
 		Email:          email,
+		UpdateName:     updateName,
+		Name:           newName,
+		UpdateTitle:updateTitle,
```
Comment (Carmendelope) [inline internal/app/cli/users.go]: Have you passed the go format?

Comment (gaizkadc) [reply internal/app/cli/users.go]: Done!

File: cmd/public-api-cli/commands/users.go [hunk 2]
```diff
@@ -47,8 +47,18 @@ func init() {
...
-	_ = updateUserCmd.MarkFlagRequired("name")
+	updateUserCmd.Flags().StringVar(&title, "title", "", "New title for the user")
+	updateUserCmd.Flags().StringVar(&phone, "phone", "", "New phone for the user")
+	updateUserCmd.Flags().StringVar(&location, "location", "", "New location for the user")
+	updateUserCmd.Flags().StringVar(&lastName, "lastName", "", "New last name for the user")
+	updateUserCmd.Flags().BoolVar(&updateName, "updateName", false, "New name for the user")
+	updateUserCmd.Flags().BoolVar(&updateTitle, "updateTitle", false, "New title for the user")
+	updateUserCmd.Flags().BoolVar(&updatePhone, "updatePhone", false, "New phone for the user")
+	updateUserCmd.Flags().BoolVar(&updateLocation, "updateLocation", false, "New location for the user")
+	updateUserCmd.Flags().BoolVar(&updateLastName, "updateLastName", false, "New last name for the user")
```
Comment (dhiguero) [inline cmd/public-api-cli/commands/users.go]: We can avoid this and improve usability by not requiring `updateFieldName` flags and checking instead for the `Changed` attribute of the flag when the command is invoked.

```
u.Update( ...., location, updateUserCmd.Flag("location").Changed, ...)
```

File: cmd/public-api-cli/commands/variables.go [hunk 3]
```diff
@@ -22,6 +22,15 @@ var loginPort int
...
 var publicKeyPath string
+var title string
+var phone string
+var location string
+var lastName string
+var updateName bool
+var updateLocation bool
+var updatePhone bool
+var updateTitle bool
+var updateLastName bool
```
Comment (dhiguero) [inline cmd/public-api-cli/commands/variables.go]: remove

File: internal/pkg/entities/validator.go [hunk 4]
```diff
@@ -248,6 +249,9 @@ func ValidUpdateUserRequest(updateUserRequest *grpc_user_go.UpdateUserRequest) d
 	if updateUserRequest.Email == "" {
 		return derrors.NewInvalidArgumentError(emptyEmail)
 	}
+	if !(updateUserRequest.UpdateLocation || updateUserRequest.UpdatePhone || updateUserRequest.UpdateTitle || updateUserRequest.UpdateName || updateUserRequest.UpdateLastName) {
```
Comment (dhiguero) [inline internal/pkg/entities/validator.go]: we are missing a check for Email

File: cmd/public-api-cli/commands/users.go [hunk 5]
```diff
@@ -136,7 +146,8 @@ var updateUserCmd = &cobra.Command{
 			cliOptions.ResolveAsInt("port", nalejPort),
 			insecure, useTLS,
 			cliOptions.Resolve("cacert", caCertPath), cliOptions.Resolve("output", output), cliOptions.ResolveAsInt("labelLength", labelLength))
-		u.Update(cliOptions.Resolve("organizationID", organizationID), email, name)
+		u.Update(cliOptions.Resolve("organizationID", organizationID), email, name, photoPath, lastName, title, phone, location)
```
Comment (dhiguero) [inline cmd/public-api-cli/commands/users.go]: A check of whether the values need to be updated is missing.

File: internal/app/cli/users.go [hunk 6]
```diff
@@ -68,13 +72,24 @@ func (u *Users) Add(organizationID string, email string, password string, name s
 	defer conn.Close()
 	defer cancel()
 
+	photoBase64 := ""
+	if photoPath != "" {
+		photoBase64 = PhotoPathToBase64(photoPath)
```
Comment (dhiguero) [inline internal/app/cli/users.go]: We should be using the existing `PhotoToBase64` and check for the error.

File: internal/app/cli/users.go [hunk 7]
```diff
@@ -170,14 +185,79 @@ func (u *Users) Update(organizationID string, email string, newName string) {
...
+	if err != nil {
+		log.Error().Err(err).Msg("cannot read image")
+		return ""
+	}
+
+	return base64.StdEncoding.EncodeToString(photoBytes)
+}
+
+// ValidateImage validates that the image is jpg or png and wights under 1 MB
+func ValidateImage(photoPath string) {
```
Comment (dhiguero) [inline internal/app/cli/users.go]: Call `ValidateImage` from the existing `PhotoToBase64`

File: internal/pkg/server/users/manager.go [hunk 8]
```diff
@@ -71,10 +75,21 @@ func (m *Manager) Add(addUserRequest *grpc_public_api_go.AddUserRequest) (*grpc_
...
+		Name:                 added.Name,
+		PhotoBase64:          "",
+		MemberSince:          0,
+		RoleId:               "",
+		RoleName:             added.RoleName,
+		LastName:             "",
+		Title:                "",
+		LastLogin:            0,
+		Phone:                "",
+		Location:             "",
```
Comment (dhiguero) [inline internal/pkg/server/users/manager.go]: Some fields are missing.

File: internal/pkg/server/users/manager.go [hunk 9]
```diff
@@ -71,10 +75,21 @@ func (m *Manager) Add(addUserRequest *grpc_public_api_go.AddUserRequest) (*grpc_
...
+		XXX_NoUnkeyedLiteral: struct{}{},
```
Comment (dhiguero) [inline internal/pkg/server/users/manager.go]:
```suggestion

```

File: internal/pkg/server/users/manager.go [hunk 10]
```diff
@@ -71,10 +75,21 @@ func (m *Manager) Add(addUserRequest *grpc_public_api_go.AddUserRequest) (*grpc_
...
+		XXX_unrecognized:     nil,
```
Comment (dhiguero) [inline internal/pkg/server/users/manager.go]:
```suggestion
```

File: internal/pkg/server/users/manager.go [hunk 11]
```diff
@@ -71,10 +75,21 @@ func (m *Manager) Add(addUserRequest *grpc_public_api_go.AddUserRequest) (*grpc_
...
+		XXX_sizecache:        0,
```
Comment (dhiguero) [inline internal/pkg/server/users/manager.go]:
```suggestion
```

File: internal/pkg/server/users/manager.go [hunk 12]
```diff
@@ -103,7 +118,7 @@ func (m *Manager) List(organizationID *grpc_organization_go.OrganizationId) (*gr
 	}
 	users := make([]*grpc_public_api_go.User, 0)
 	for _, u := range list.Users {
-		if !u.Internal {
+		if !u.InternalRole {
 			toAdd := &grpc_public_api_go.User{
 				OrganizationId: u.OrganizationId,
```
Comment (dhiguero) [inline internal/pkg/server/users/manager.go]: Some fields are missing

File: internal/app/cli/users.go [hunk 13]
```diff
@@ -74,7 +70,7 @@ func (u *Users) Add(organizationID string, email string, password string, name s
 
 	photoBase64 := ""
 	if photoPath != "" {
-		photoBase64 = PhotoPathToBase64(photoPath)
+		photoBase64, _ = PhotoPathToBase64(photoPath)
```
Comment (dhiguero) [inline internal/app/cli/users.go]: Error check is missing

**Pull request merged**